eater/models/partner.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2020 Coop IT Easy SCRL fs
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    eater = fields.Selection(
+        [("eater", "Eater"), ("worker_eater", "Worker and Eater")],
+        string="Eater/Worker",
+    )
+    child_eater_ids = fields.One2many(
+        "res.partner",
+        "parent_eater_id",
+        domain=[("eater", "=", "eater")],
+    )
+    parent_eater_id = fields.Many2one(
+        "res.partner",
+        string="Parent Worker",
+        readonly=True,
+        domain=[("eater", "=", "worker_eater")],
+    )
+
+    @api.constrains("eater", "parent_eater_id")
+    def _check_parent_is_worker(self):
+        """The parent of an eater must be a worker_eater, and worker_eaters
+        cannot have parents.
+        """
+        for partner in self:
+            parent = partner.parent_eater_id
+            if partner.eater == "eater" and parent:
+                if parent.eater != "worker_eater":
+                    raise ValidationError(
+                        _(
+                            "{0} cannot be the parent of {1} because the parent"
+                            " must be a worker."
+                        ).format(parent.name, partner.name)
+                    )
+            if partner.eater == "worker_eater" and parent:
+                raise ValidationError(
+                    _(
+                        "%s cannot have a parent worker because they are"
+                        " themselves a worker."
+                    )
+                    % partner.name
+                )
+
+    def write(self, values):
+        if values.get("parent_eater_id"):
+            for partner in self:
+                if (
+                    partner.parent_eater_id
+                    and partner.parent_eater_id.id != values.get("parent_eater_id")
+                ):
+                    raise ValidationError(
+                        _(
+                            "You are trying to assign %(eater)s as an eater to"
+                            " a new partner, but this eater is already"
+                            " assigned to %(old_partner)s. Please remove the link"
+                            " before creating a new one."
+                        )
+                        % {
+                            "eater": partner.name,
+                            "old_partner": partner.parent_eater_id.name,
+                        }
+                    )
```

Comment:
What is the use case here?

Reply:
You can't be the eater of two workers.

Reply:
But this would change the parent eater ("worker"), no? So after the operation, you still only have one parent.

Reply:
Apparently, 8 years ago, this field was a many2many => removing the code.